### ISSUE TITLE
AI-Powered Image Restoration and Enhancement Tool

### ISSUE THREAD

**Issue by ramana2074** [abhisheks008/DL-Simplified#1003]
---
name: DL-Simplified Issue Template
about: This is the issue template that is going to be used while creating any issues
  in this repo.
title: ''
labels: ''
assignees: ''

---

### Deep Learning Simplified Repository (Proposing new issue)
:red_circle: **Project Title** : </br> AI-Powered Image Restoration and Enhancement Tool
:red_circle: **Aim** : </br> This project aims to develop an AI-Powered Image Restoration Tool that revitalizes low-quality historical photos by upscaling, denoising, and adding realistic color. 
:red_circle: **Dataset** : </br> Pretrained model
:red_circle: **Approach** :  Using advanced deep learning techniques, it transforms degraded images into vibrant, high-quality visuals while preserving their historical context. This tool is perfect for heritage conservation, family archiving, and historical documentation. 
SRCNN (Super-Resolution Convolutional Neural Network): For image upscaling
DnCNN: For image denoising
Pre-trained colorization models (e.g., U-Net): For adding color to grayscale images

**********************************************************************************
### 📍 **Follow the Guidelines to Contribute in the Project :**
* You need to create a separate folder named as the Project Title.
* Inside that folder, there will be four main components.
   * Images - To store the required images.
   * Dataset - To store the dataset or, information/source about the dataset.
   * Model - To store the machine learning model you've created using the dataset.
   * `requirements.txt` - This file will contain the required packages/libraries to run the project in other machines.
* Inside the `Model` folder, the `README.md` file must be filled up properly, with proper visualizations and conclusions.

***********************************************************************
:red_circle::yellow_circle: **Points to Note :**

- The issues will be assigned on a first come first serve basis, 1 Issue == 1 PR.
- "Issue Title" and "PR Title should be the same. Include issue number along with it.
- Follow Contributing Guidelines & Code of Conduct before start Contributing.

***********************************************************************
:white_check_mark: **To be Mentioned while taking the issue :**
- Full name : Venkata Ramana Billana
- GitHub Profile Link :  https://github.com/ramana2074
- Email ID : billanavenkataramana@gmail.com
- Participant ID (if applicable):
- Approach for this Project : Using advanced deep learning techniques, it transforms degraded images into vibrant, high-quality visuals while preserving their historical context. This tool is perfect for heritage conservation, family archiving, and historical documentation. 
- What is your participant role? (Mention the Open Source program) - SWOC

*************************************************************
Happy Contributing 🚀 

All the best. Enjoy your open source journey ahead. 😎

**Comment by abhisheks008** [comment]
Hi @ramana2074 issue assigned to you.

**Comment by Sonmale25** [comment]
### Full name:
Pranav Prakash Sonmale

### GitHub Profile Link:
https://github.com/Sonmale25

### Email ID:
sonmalepranav@gmail.com

### What is your participant role?
SSOC ’25 Contributor

**### Proposed Approach**
I will build a pipeline that restores, denoises, upscales, and colorizes historical images using state‑of‑the‑art convolutional architectures:

**### Model Components & Order**

Denoising: DnCNN to remove sensor noise and scratches.

Super‑resolution (Upscaling): SRCNN to increase resolution by 2× or 4×.

Colorization: U‑Net‑based pre‑trained colorization network to add realistic color while preserving historical detail.

### **Tech Stack**

**Framework:** Python 3.9+ with PyTorch for model development.

**Dependencies:**

torch, torchvision

numpy, opencv-python for image I/O & preprocessing

matplotlib or seaborn for visualizations

Pillow for format conversions

### **Dataset & Pretrained Weights**

DnCNN: use official pretrained weights trained on BSD400.

SRCNN: use publicly available weights trained on DIV2K dataset.

Colorization U‑Net: leverage “Richard Zhang et al.” pretrained model from [GitHub/colorful‑image‑colorization].

Custom Fine‑Tuning (Optional): allow fine‑tuning on user‑provided small historical‑photo datasets (e.g. Yorick’s “Historical Photo Restorations” set) for domain adaptation.

### **Inference Pipeline**

Load an input image → apply DnCNN → feed denoised image to SRCNN → pass upscaled output to U‑Net colorizer → save final PNG/JPEG.

Provide a Jupyter notebook (Model/inference_demo.ipynb) demonstrating end‑to‑end results on sample images.

### **Evaluation & Visualization**

Compute PSNR and SSIM against available ground‑truth restorations (if any).

In Model/README.md, include side‑by‑side before/after comparisons, quantitative metrics, and brief conclusions on restoration quality.

Looking forward to contributing—please assign this

**Comment by GoliJyothish** [comment]
Hi, I would like to work on this issue. I am a beginner in deep learning and computer vision. Please assign this to me.

**Comment by abhisheks008** [comment]
> Hi, I would like to work on this issue. I am a beginner in deep learning and computer vision. Please assign this to me.

Hi @GoliJyothish thanks for showing interest. Can you please elaborate your approach for solving this issue.
- Models details (at least 4 models)
- Accuracy parameters of the models
- EDA of the dataset/structure of the dataset
- Reason for the chosen models

**Comment by GoliJyothish** [comment]
Hi @abhisheks008! Here is my approach for this issue:
1. Models (4 models):
DnCNN — for image denoising
SRCNN — for super resolution
U-Net — for image colorization
ESRGAN — for enhanced super resolution

2. Accuracy Parameters:
PSNR (Peak Signal-to-Noise Ratio) — measures image quality
SSIM (Structural Similarity Index) — measures structural similarity
MSE (Mean Squared Error) — measures pixel-level loss

3. Dataset & EDA:
BSD300 dataset for denoising
Set5/Set14 for super resolution
EDA will include: image size distribution, noise level analysis, before/after comparisons

4. Reason for chosen models:
DnCNN and SRCNN are lightweight and well-proven for restoration tasks
U-Net is best for pixel-wise tasks like colorization
ESRGAN gives best visual quality for super resolution

Looking forward to your approval!

**Comment by abhisheks008** [comment]
Hi @GoliJyothish can you clarify a bit on the ESRGAN model? How this is effective here with the dataset? Also can you share the dataset details, any URL or source?

**Comment by GoliJyothish** [comment]
Hi @abhisheks008! 

**1. ESRGAN effectiveness with BSD300:**
ESRGAN (Enhanced Super Resolution GAN) is effective here because BSD300 contains 
high-resolution natural images (animals, landscapes, people) which are ideal for 
training super resolution models. We downsample these images to 64x64 and use 
ESRGAN to reconstruct them back to 256x256 (4x upscale). BSD300's diverse textures 
and edges make it a strong benchmark dataset for evaluating super resolution quality 
using PSNR and SSIM metrics.

**2. Dataset Details:**
- **Name:** BSD300 (Berkeley Segmentation Dataset 300)
- **Source:** UC Berkeley Computer Vision Group
- **URL:** https://www2.eecs.berkeley.edu/Research/Projects/CS/vision/bsds/
- **Size:** 300 images (200 train / 100 test)
- **Image size:** 321×481 pixels (natural images)
- **Format:** JPEG
- **Usage:** Widely used benchmark for denoising, super resolution, and segmentation tasks

**Comment by abhisheks008** [comment]
You can proceed with this issue. @GoliJyothish assigned to you.
